src/test/groovy/com/stormpath/tck/register/RegisterIT.groovy

```diff
@@ -0,0 +1,576 @@
+        jsonAsMap.put("surname", accountSurname)
+        jsonAsMap.put("username", accountUsername)
+
+        String createdHref =
+            given()
+                .accept(ContentType.JSON)
+                .body(jsonAsMap)
+            .when()
+                .post(registerRoute)
+            .then()
+                .statusCode(200)
+                .contentType(ContentType.JSON)
+                .body("size()", is(1))
+                .body("account.href", not(isEmptyOrNullString()))
+                .body("account.username", is(accountUsername))
+                .body("account.modifiedAt", not(isEmptyOrNullString()))
+                .body("account.status", equalToIgnoringCase("ENABLED"))
+                .body("account.createdAt", not(isEmptyOrNullString()))
+                .body("account.email", is("json-$accountEmail".toString()))
+                .body("account.middleName", is(accountMiddleName))
+                .body("account.surname", is(accountSurname))
+                .body("account.givenName", is(accountGivenName))
+                .body("account.fullName", is("$accountGivenName $accountMiddleName $accountSurname".toString()))
+            .extract()
+                .path("account.href")
+
+        deleteOnClassTeardown(createdHref)
+    }
+
+    /**
+     * Serve a default HTML form for request type text/html
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/180">#180</a>
+     * @throws Exception
+     */
+    @Test
+    public void servesRegisterForm() throws Exception {
+
+        Response response =
+            given()
+                .accept(ContentType.HTML)
+            .when()
+                .get(registerRoute)
+            .then()
+                .statusCode(200)
+                .contentType(ContentType.HTML)
+            .extract()
+                .response()
+
+        XmlPath doc = getHtmlDoc(response)
+
+        Node submitButton = findTagWithAttribute(doc.getNodeChildren("html.body"), "button", "type", "submit")
+        assertEquals(submitButton.value(), "Create Account")
+    }
+
+    /** Default form should include fields ordered by fieldOrder
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/181">#181</a>
+     * @throws Exception
+     */
+    @Test
+    public void formShouldContainFieldsOrderedByFieldOrder() throws Exception {
+
+        // Todo: CSRF support
+
+        Response response =
+            given()
+                .accept(ContentType.HTML)
+            .when()
+                .get(registerRoute)
+            .then()
+                .statusCode(200)
+                .contentType(ContentType.HTML)
+            .extract()
+                .response()
+
+        XmlPath doc = getHtmlDoc(response)
+        List<Node> fields = findTags(doc.getNodeChildren("html.body"), "input")
+
+        // From default configuration
+        assertEquals(fields.get(0).attributes().get("name"), "givenName")
```

Comment:
Can you grab the fields, and iterate through them to make sure that they're in the right order (but not necessarily index 0)?